### PR TITLE
A few more changes to planner GUI before removing...

### PR DESCRIPTION
**Pull request by bminortx**
< feeling sheepish >
Splits PathPlannerTest into two programs, one for solving paths, and one for interpolating bezier commands.

### CODE REVIEW

File: Applications/Examples/PathPlannerTest/PathPlannerInterpolation.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  // -0.4 0.6 5 -2 -0.2 0.5 0 0 1.0737 0 2.1474 0 3.2331  0.10952    4.3 -0.011462 5.3669 -0.13245
+  // -0.4 0.6 5 -2    0   1 0 0  1.077 0 2.1541 0 3.4001 -0.73386 4.3921  -0.31445 5.3841  0.10497
+  double solution[] =
+      {-0.4, 0.6, 5, -2, -0.1, 0.75, 0, 0, 1.07535, 0, 2.15075, 0, 3.3166,
+       -0.3121699, 4.34605, -0.162956, 5.3755, -0.01374};
+      // {-0.4, 0.6, 5, -2, -0.4, 0.5, 0, 0, 1.0699, 0, 2.1397, 0, 3.1377,
+      //  -0.68632, 4.1671, -0.97782, 5.1965, -1.2693};
+  start.push_back(0);
+  start.push_back(0);
+  start.push_back(solution[0]);
```
Comment (chachi) [inline Applications/Examples/PathPlannerTest/PathPlannerInterpolation.cpp]: Initializer lists

File: Applications/Examples/PathPlannerTest/PathPlannerSolver.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  // Boilerplate stuff
+  PathPlannerTest* sim = new PathPlannerTest();
+  std::string name = "Sim0";
+  if (argc == 2) {
+    name = argv[1];
+  }
+  sim->sim_planner_name_ = name;
+  std::string number = sim->GetNumber(sim->sim_planner_name_);
+  URDF_Parser* parser = new URDF_Parser(0);
```
Comment (chachi) [inline Applications/Examples/PathPlannerTest/PathPlannerSolver.cpp]: memory leak

File: Applications/Examples/PathPlannerTest/PathPlannerTest.cpp [hunk 3]
```diff
@@ -182,6 +176,72 @@ void PathPlannerTest::SampleTrajectory(pb::PlannerPolicyMsg* policy){
...
+  bool success = false;
+  int count = 0;
+  int max_count = 1000;
+  ApplyVelocitesFunctor5d func(car_model_.get(), Eigen::Vector3d::Zero(), NULL);
+  func.SetNoDelay(true);
+  MotionSample sample;
+  car_model_->SetState(0, start_state_);
+  GLWayPoint* a = &planner_gui_.GetWaypoint(0)->m_Waypoint;
+  GLWayPoint* b = &planner_gui_.GetWaypoint(1)->m_Waypoint;
+  VehicleState st(Sophus::SE3d(a->GetPose4x4_po()),a->GetVelocity(),0);
```
Comment (chachi) [inline Applications/Examples/PathPlannerTest/PathPlannerTest.cpp]: Spaces after `,`

File: Applications/Examples/PathPlannerTest/PathPlannerTest.cpp [hunk 4]
```diff
@@ -182,6 +176,72 @@ void PathPlannerTest::SampleTrajectory(pb::PlannerPolicyMsg* policy){
...
+                                       Eigen::VectorXd x_values,
+                                       Eigen::VectorXd y_values) {
+  bool success = false;
+  int count = 0;
+  int max_count = 1000;
+  ApplyVelocitesFunctor5d func(car_model_.get(), Eigen::Vector3d::Zero(), NULL);
+  func.SetNoDelay(true);
+  MotionSample sample;
+  car_model_->SetState(0, start_state_);
+  GLWayPoint* a = &planner_gui_.GetWaypoint(0)->m_Waypoint;
```
Comment (chachi) [inline Applications/Examples/PathPlannerTest/PathPlannerTest.cpp]: It doesn't semantically make a difference, but if you don't want to have to use pointer notation, you could do this:

`GLWayPoint& a = planner_gui_.GetWaypoint(0)->m_Waypoint;`

Then in the following lines you could just say `a.GetPose4x4_po()`. Again, no difference in behavior (probably gets compiled to the exact same thing), just want you to know your options.

File: Applications/Examples/PathPlannerTest/PathPlannerTest.cpp [hunk 5]
```diff
@@ -182,6 +176,72 @@ void PathPlannerTest::SampleTrajectory(pb::PlannerPolicyMsg* policy){
...
+  // Set our Bezier boundary problem's values
+  // This means that it won't try to solve for the trajectory itself.
+  // There's a nicer way to do this, I'm sure, but I haven't programmed it.
+  BezierBoundaryProblem* spline = problem.GetBezierProblem();
+  spline->x_values_ = x_values;
+  spline->y_values_ = y_values;
+  spline->has_given_points_ = true;
+  Eigen::Vector6d spline_vec;
+  for (int ii = 0; ii<6; ii++) {
+    spline_vec<<spline->x_values_(ii), spline->y_values_(ii), 0, 0, 0, 0;
```
Comment (chachi) [inline Applications/Examples/PathPlannerTest/PathPlannerTest.cpp]: Spacing around `<<`

File: Applications/Examples/PathPlannerTest/PathPlannerTest.cpp [hunk 6]
```diff
@@ -182,6 +176,72 @@ void PathPlannerTest::SampleTrajectory(pb::PlannerPolicyMsg* policy){
...
+    planner_gui_.AddSplinePoints(spline_vec, ii);
+  }
+  local_planner_.InitializeLocalProblem(problem, 0, NULL, eCostPoint);
+  for (int ii = 0; ii<200; ii++) {
```
Comment (chachi) [inline Applications/Examples/PathPlannerTest/PathPlannerTest.cpp]: Spacing around `<` and others below

File: Applications/PathPlanner/PathPlanner.cpp [hunk 7]
```diff
@@ -170,14 +162,14 @@ void PathPlanner::GroundStates() {
   Sophus::SE3d pose = start_state_.m_dTwv;
   if (car_model_->RayCast(pose.translation(), GetBasisVector(pose,2)*10,
                           dIntersect, true, 0)) {
-    dIntersect(2) = dIntersect(2)+.12;
+    dIntersect(2) = dIntersect(2)+.15;
```
Comment (chachi) [inline Applications/PathPlanner/PathPlanner.cpp:159]: Spacing around `+`

**Comment by bminortx** [comment]
@chachi, when you're done with the jenkins build, could you check this PR over? I'm at a good merging point.

**Comment by chachi** [comment]
@bminortx Just a couple small suggestions. Otherwise, looks good to me.

**Comment by chachi** [comment]
I added just a couple fixes to hopefully make the build pass again.

**Comment by chachi** [comment]
testingtesting

**Pull request merged**